### PR TITLE
fix(web): extra spaces in HTML attribute

### PR DESCRIPTION
**Pull request by kecrily**
Remove extra spaces in class attributes caused by conditional statements

Example:

```html
<div class="{{if .Source.IsLDAP}}required{{end}} field">
```

When `IsLDAP` is false, will render `<div class=" field">` not `<div class="field">`. So should put space in conditional statements.

---

Changes are made by VS Code regex replace and are checked manually.

- `class="\{\{if (.*)\{\{end\}\} ` -> `class="{{if $1 {{end}\}`
- `class="(.*) \{\{if (.*)\}\}(.*)\{\{end\}\}` -> `class="$1{{if $2}} $3{{end}}`

### CODE REVIEW

File: templates/repo/issue/view_content/sidebar.tmpl [hunk 1]
```diff
@@ -23,7 +23,7 @@
 					{{if .Reviewers}}
 						{{range .Reviewers}}
 							{{if .User}}
-								<a class="{{if not .CanChange}}ui tooltip{{end}} item {{if .Checked}} checked {{end}} {{if not .CanChange}}ban-change{{end}}" href="#" data-id="{{.ItemID}}" data-id-selector="#review_request_{{.ItemID}}" {{if not .CanChange}} data-content="{{$.locale.Tr "repo.issues.remove_request_review_block"}}"{{end}}>
+								<a class="{{if not .CanChange}}ui tooltip{{end}} item {{if .Checked}} checked  {{end}}{{if not .CanChange}}ban-change{{end}}" href="#" data-id="{{.ItemID}}" data-id-selector="#review_request_{{.ItemID}}" {{if not .CanChange}} data-content="{{$.locale.Tr "repo.issues.remove_request_review_block"}}"{{end}}>
```
Comment (delvh) [inline templates/repo/issue/view_content/sidebar.tmpl]:
```suggestion
								<a class="item{{if not .CanChange}} ui tooltip{{end}}{{if .Checked}} checked{{end}}{{if not .CanChange}} ban-change{{end}}" href="#" data-id="{{.ItemID}}" data-id-selector="#review_request_{{.ItemID}}" {{if not .CanChange}} data-content="{{$.locale.Tr "repo.issues.remove_request_review_block"}}"{{end}}>
```

File: templates/repo/issue/view_content/sidebar.tmpl [hunk 2]
```diff
@@ -37,7 +37,7 @@
 						<div class="ui divider"></div>
 						{{range .TeamReviewers}}
 							{{if .Team}}
-								<a class="{{if not .CanChange}}ui tooltip{{end}} item {{if .Checked}} checked {{end}} {{if not .CanChange}}ban-change{{end}}" href="#" data-id="{{.ItemID}}" data-id-selector="#review_request_team_{{.Team.ID}}" {{if not .CanChange}} data-content="{{$.locale.Tr "repo.issues.remove_request_review_block"}}"{{end}}>
+								<a class="{{if not .CanChange}}ui tooltip{{end}} item {{if .Checked}} checked  {{end}}{{if not .CanChange}}ban-change{{end}}" href="#" data-id="{{.ItemID}}" data-id-selector="#review_request_team_{{.Team.ID}}" {{if not .CanChange}} data-content="{{$.locale.Tr "repo.issues.remove_request_review_block"}}"{{end}}>
```
Comment (delvh) [inline templates/repo/issue/view_content/sidebar.tmpl]:
```suggestion
								<a class="item{{if not .CanChange}} ui tooltip{{end}}{{if .Checked}} checked{{end}}{{if not .CanChange}} ban-change{{end}}" href="#" data-id="{{.ItemID}}" data-id-selector="#review_request_team_{{.Team.ID}}"{{if not .CanChange}} data-content="{{$.locale.Tr "repo.issues.remove_request_review_block"}}"{{end}}>
```

File: templates/repo/issue/view_content/sidebar.tmpl [hunk 3]
```diff
@@ -124,12 +124,12 @@
 				<div class="no-select item">{{.locale.Tr "repo.issues.new.clear_labels"}}</div>
 				{{if or .Labels .OrgLabels}}
 					{{range .Labels}}
-						<a class="{{if .IsChecked}}checked{{end}} item" href="#" data-id="{{.ID}}" data-id-selector="#label_{{.ID}}"><span class="octicon-check {{if not .IsChecked}}invisible{{end}}">{{svg "octicon-check"}}</span><span class="label color" style="background-color: {{.Color}}"></span> {{.Name | RenderEmoji}}
+						<a class="{{if .IsChecked}}checked {{end}}item" href="#" data-id="{{.ID}}" data-id-selector="#label_{{.ID}}"><span class="octicon-check {{if not .IsChecked}}invisible{{end}}">{{svg "octicon-check"}}</span><span class="label color" style="background-color: {{.Color}}"></span> {{.Name | RenderEmoji}}
```
Comment (delvh) [inline templates/repo/issue/view_content/sidebar.tmpl]:
```suggestion
						<a class="{{if .IsChecked}}checked {{end}}item" href="#" data-id="{{.ID}}" data-id-selector="#label_{{.ID}}"><span class="octicon-check{{if not .IsChecked}} invisible{{end}}">{{svg "octicon-check"}}</span><span class="label color" style="background-color: {{.Color}}"></span> {{.Name | RenderEmoji}}
```

File: templates/repo/issue/view_content/sidebar.tmpl [hunk 4]
```diff
@@ -124,12 +124,12 @@
...
 						{{if .Description}}<br><small class="desc">{{.Description | RenderEmoji}}</small>{{end}}</a>
 					{{end}}
 					<div class="ui divider"></div>
 					{{range .OrgLabels}}
-						<a class="{{if .IsChecked}}checked{{end}} item" href="#" data-id="{{.ID}}" data-id-selector="#label_{{.ID}}"><span class="octicon-check {{if not .IsChecked}}invisible{{end}}">{{svg "octicon-check"}}</span><span class="label color" style="background-color: {{.Color}}"></span> {{.Name | RenderEmoji}}
+						<a class="{{if .IsChecked}}checked {{end}}item" href="#" data-id="{{.ID}}" data-id-selector="#label_{{.ID}}"><span class="octicon-check {{if not .IsChecked}}invisible{{end}}">{{svg "octicon-check"}}</span><span class="label color" style="background-color: {{.Color}}"></span> {{.Name | RenderEmoji}}
```
Comment (delvh) [inline templates/repo/issue/view_content/sidebar.tmpl]:
```suggestion
						<a class="{{if .IsChecked}}checked {{end}}item" href="#" data-id="{{.ID}}" data-id-selector="#label_{{.ID}}"><span class="octicon-check{{if not .IsChecked}} invisible{{end}}">{{svg "octicon-check"}}</span><span class="label color" style="background-color: {{.Color}}"></span> {{.Name | RenderEmoji}}
```

File: templates/repo/issue/list.tmpl [hunk 5]
```diff
@@ -70,7 +70,7 @@
 							</div>
 							<a class="item" href="{{$.Link}}?q={{$.Keyword}}&type={{$.ViewType}}&sort={{$.SortType}}&state={{$.State}}&labels={{.SelectLabels}}&assignee={{$.AssigneeID}}&poster={{$.PosterID}}">{{.locale.Tr "repo.issues.filter_milestone_no_select"}}</a>
 							{{range .Milestones}}
-								<a class="{{if $.MilestoneID}}{{if eq $.MilestoneID .ID}}active selected{{end}}{{end}} item" href="{{$.Link}}?type={{$.ViewType}}&sort={{$.SortType}}&state={{$.State}}&labels={{$.SelectLabels}}&milestone={{.ID}}&assignee={{$.AssigneeID}}&poster={{$.PosterID}}">{{.Name}}</a>
+								<a class="{{if $.MilestoneID}}{{if eq $.MilestoneID .ID}}active selected{{end}} {{end}}item" href="{{$.Link}}?type={{$.ViewType}}&sort={{$.SortType}}&state={{$.State}}&labels={{$.SelectLabels}}&milestone={{.ID}}&assignee={{$.AssigneeID}}&poster={{$.PosterID}}">{{.Name}}</a>
```
Comment (delvh) [inline templates/repo/issue/list.tmpl]:
```suggestion
								<a class="{{if and $.MilestoneID (eq $.MilestoneID .ID)}}active selected {{end}}item" href="{{$.Link}}?type={{$.ViewType}}&sort={{$.SortType}}&state={{$.State}}&labels={{$.SelectLabels}}&milestone={{.ID}}&assignee={{$.AssigneeID}}&poster={{$.PosterID}}">{{.Name}}</a>
```

**Review by delvh — COMMENTED** [review]
Your automatic check seems not to have included all cases.

Should we also include the other case `char {{<something>}}class` in this PR, or do we open another PR for it?

**Comment by silverwind** [comment]
If you are really sure about your regexes, you could put them below this one as a sed replacement for auto-formatting:

https://github.com/go-gitea/gitea/blob/0b993a0d04c9e542f1584ce00eab7ee4af0321d2/Makefile#L271

File: templates/repo/issue/view_content/sidebar.tmpl (hunk 4 above)
Comment (kecrily) [reply templates/repo/issue/view_content/sidebar.tmpl]: covered by `class="(.*) \{\{if (.*)\}\}(.*)\{\{end\}\}` -> `class="$1{{if $2}} $3{{end}}`

**Comment by kecrily** [comment]
In fact I'm not very confident in my regex. I was afraid it would inadvertently destroy something.

File: templates/admin/auth/edit.tmpl [hunk 6]
```diff
@@ -114,10 +114,10 @@
 					<div class="inline field">
 						<div class="ui checkbox">
 							<label><strong>{{.locale.Tr "admin.auths.enable_ldap_groups"}}</strong></label>
-							<input type="checkbox" name="groups_enabled" class="js-ldap-group-toggle" {{if $cfg.GroupsEnabled}}checked{{end}}>
+							<input type="checkbox" name="groups_enabled" class="js-ldap-group-toggle"{{if $cfg.GroupsEnabled}} checked{{end}}>
```
Comment (kecrily) [inline templates/admin/auth/edit.tmpl:117]:
```suggestion
							<input type="checkbox" name="groups_enabled" class="js-ldap-group-toggle{{if $cfg.GroupsEnabled}} checked{{end}}">
```

The double quotes are in the wrong place here, and I'm not sure if it should be solved in this PR, or if it should be solved in a separate PR.

File: templates/admin/auth/edit.tmpl [hunk 7]
```diff
@@ -114,10 +114,10 @@
...
 						</div>
 					</div>
-					<div id="ldap-group-options" class="ui segment secondary" {{if not $cfg.GroupsEnabled}}hidden{{end}}>
+					<div id="ldap-group-options" class="ui segment secondary"{{if not $cfg.GroupsEnabled}} hidden{{end}}>
```
Comment (kecrily) [inline templates/admin/auth/edit.tmpl:120]:
```suggestion
					<div id="ldap-group-options" class="ui segment secondary{{if not $cfg.GroupsEnabled}} hidden{{end}}">
```

File: templates/user/dashboard/milestones.tmpl [hunk 8]
```diff
@@ -27,7 +27,7 @@
 									{{- end -}}
 								]&sort={{$.SortType}}&state={{$.State}}&q={{$.Keyword}}" title="{{.FullName}}">
 								<span class="text truncate">{{$Repo.FullName}}</span>
-								<div class="ui {{if $.IsShowClosed}}red{{else}}green{{end}} label">{{index $.Counts $Repo.ID}}</div>
+								<div class="ui{{if $.IsShowClosed}} red{{else}} green{{end}} label">{{index $.Counts $Repo.ID}}</div>
```
Comment (silverwind) [inline templates/user/dashboard/milestones.tmpl]: In such if+else cases, it's more efficient to keep the space before as it saves 1 byte in the template.

File: templates/repo/settings/protected_branch.tmpl [hunk 9]
```diff
@@ -146,10 +146,10 @@
 								{{range $.branch_status_check_contexts}}
 									<tr><td>
 										<span class="ui checkbox">
-											<input class="enable-whitelist" name="status_check_contexts" value="{{.}}" type="checkbox" {{if $.is_context_required}}{{if call $.is_context_required .}}checked{{end}}{{end}}>
+											<input class="enable-whitelist" name="status_check_contexts" value="{{.}}" type="checkbox"{{if $.is_context_required and (call $.is_context_required .)}} checked{{end}}>
 										</span>
 										{{.}}
-										{{if $.is_context_required}}{{if call $.is_context_required .}}<div class="ui label right">Required</div>{{end}}{{end}}
+										{{if $.is_context_required and (call $.is_context_required .)}}<div class="ui label right">Required</div>{{end}}
```
Comment (silverwind) [inline templates/repo/settings/protected_branch.tmpl]: This is invalid syntax, you need to `{{if and <conditions,...>}}`

File: templates/repo/issue/list.tmpl [hunk 10]
```diff
@@ -70,13 +70,13 @@
 							</div>
 							<a class="item" href="{{$.Link}}?q={{$.Keyword}}&type={{$.ViewType}}&sort={{$.SortType}}&state={{$.State}}&labels={{.SelectLabels}}&assignee={{$.AssigneeID}}&poster={{$.PosterID}}">{{.locale.Tr "repo.issues.filter_milestone_no_select"}}</a>
 							{{range .Milestones}}
-								<a class="{{if $.MilestoneID}}{{if eq $.MilestoneID .ID}}active selected{{end}}{{end}} item" href="{{$.Link}}?type={{$.ViewType}}&sort={{$.SortType}}&state={{$.State}}&labels={{$.SelectLabels}}&milestone={{.ID}}&assignee={{$.AssigneeID}}&poster={{$.PosterID}}">{{.Name}}</a>
+								<a class="{{if $.MilestoneID and (eq $.MilestoneID .ID)}}active selected {{end}}item" href="{{$.Link}}?type={{$.ViewType}}&sort={{$.SortType}}&state={{$.State}}&labels={{$.SelectLabels}}&milestone={{.ID}}&assignee={{$.AssigneeID}}&poster={{$.PosterID}}">{{.Name}}</a>
```
Comment (silverwind) [inline templates/repo/issue/list.tmpl]: also invalid syntax.

File: templates/user/dashboard/milestones.tmpl (hunk 8 above)
Comment (kecrily) [reply templates/user/dashboard/milestones.tmpl]: One is to reduce one character on the development side and one is to reduce one character on the user side. I prefer to reduce one character on the user side, but it's not a big deal, it's up to you

Comment (silverwind) [reply templates/user/dashboard/milestones.tmpl]: The rendered outcome is the same because one of the if branches will be hit in an if-else. Not a big deal but I guess I prefer the 1 byte reduction.

**Comment by silverwind** [comment]
Need to rebase.

**Comment by silverwind** [comment]
Test failures do seem causes by the changes, but it's not immediately clear to my why.

**Comment by zeripath** [comment]
> Test failures do seem related, but it's not immediately clear to my why.

```
--- FAIL: TestPullCompare (0.43s)
    testlogger.go:78: 2022/11/20 15:36:35 ...eb/routing/logger.go:99:func1() [I] [637a4983] router: completed GET /user2/repo1/pulls for , 200 OK in 98.0ms @ repo/issue.go:408(repo.Issues)
    pull_compare_test.go:27: 
        	Error Trace:	/drone/src/integration_test.go:337
        	            				/drone/src/integration_test.go:166
        	            				/drone/src/pull_compare_test.go:27
        	Error:      	Not equal: 
        	            	expected: 200
        	            	actual  : 404
        	Test:       	TestPullCompare
        	Messages:   	Request: GET /user2/repo1/compare/master...master%20
    pull_compare_test.go:27: Response: Not found.

```     

In particular:

```
Messages:   	Request: GET /user2/repo1/compare/master...master%20
```

I suspect the changes have broken the url somehow.

File: templates/repo/diff/compare.tmpl [hunk 11]
```diff
@@ -70,7 +70,7 @@
 				</div>
 				<div class="scrolling menu reference-list-menu base-branch-list">
 					{{range .Branches}}
-						<div class="item {{if eq $.BaseBranch .}}selected{{end}}" data-url="{{$.RepoLink}}/compare/{{PathEscapeSegments .}}{{$.CompareSeparator}}{{if not $.PullRequestCtx.SameRepo}}{{PathEscape $.HeadUser.Name}}/{{PathEscape $.HeadRepo.Name}}:{{end}}{{PathEscapeSegments $.HeadBranch}}">{{$BaseCompareName}}:{{.}}</div>
+						<div class="item{{if eq $.BaseBranch .}}selected{{end}}" data-url="{{$.RepoLink}}/compare/{{PathEscapeSegments .}}{{$.CompareSeparator}}{{if not $.PullRequestCtx.SameRepo}}{{PathEscape $.HeadUser.Name}}/{{PathEscape $.HeadRepo.Name}} :{{end}}{{PathEscapeSegments $.HeadBranch}}">{{$BaseCompareName}}:{{.}}</div>
```
Comment (zeripath) [inline templates/repo/diff/compare.tmpl]:
```suggestion
						<div class="item{{if eq $.BaseBranch .}} selected{{end}}" data-url="{{$.RepoLink}}/compare/{{PathEscapeSegments .}}{{$.CompareSeparator}}{{if not $.PullRequestCtx.SameRepo}}{{PathEscape $.HeadUser.Name}}/{{PathEscape $.HeadRepo.Name}}:{{end}}{{PathEscapeSegments $.HeadBranch}}">{{$BaseCompareName}}:{{.}}</div>
```

File: templates/repo/diff/compare.tmpl [hunk 12]
```diff
@@ -90,7 +90,7 @@
 				</div>
 				<div class="scrolling menu reference-list-menu base-tag-list" style="display: none">
 					{{range .Tags}}
-						<div class="item {{if eq $.BaseBranch .}}selected{{end}}" data-url="{{$.RepoLink}}/compare/{{PathEscapeSegments .}}{{$.CompareSeparator}}{{if not $.PullRequestCtx.SameRepo}}{{PathEscape $.HeadUser.Name}}/{{PathEscape $.HeadRepo.Name}}:{{end}}{{PathEscapeSegments $.HeadBranch}}">{{$BaseCompareName}}:{{.}}</div>
+						<div class="item{{if eq $.BaseBranch .}}selected{{end}}" data-url="{{$.RepoLink}}/compare/{{PathEscapeSegments .}}{{$.CompareSeparator}}{{if not $.PullRequestCtx.SameRepo}}{{PathEscape $.HeadUser.Name}}/{{PathEscape $.HeadRepo.Name}} :{{end}}{{PathEscapeSegments $.HeadBranch}}">{{$BaseCompareName}}:{{.}}</div>
```
Comment (zeripath) [inline templates/repo/diff/compare.tmpl]:
```suggestion
						<div class="item{{if eq $.BaseBranch .}} selected{{end}}" data-url="{{$.RepoLink}}/compare/{{PathEscapeSegments .}}{{$.CompareSeparator}}{{if not $.PullRequestCtx.SameRepo}}{{PathEscape $.HeadUser.Name}}/{{PathEscape $.HeadRepo.Name}}:{{end}}{{PathEscapeSegments $.HeadBranch}}">{{$BaseCompareName}}:{{.}}</div>
```

**Comment by zeripath** [comment]
The above two things should fix the problem but we should really go through this PR and double check that it's doing the correct thing.

**Comment by kecrily** [comment]
The issue was constructed with so many files (a mind-boggling 148) that regex were my only option. My regex isn't very good, therefore it could have resulted in an incorrect alteration.

I also made an effort to go back and revert any incorrect alterations, but it was simply too much. Inevitably, something is missed, and we can only rely on the keen eye of the reviewer.

**Comment by silverwind** [comment]
First failure on CI is odd because it's being seen outside this PR too, so I think it may be flaky/unrelated. See https://github.com/go-gitea/gitea/issues/21854.

File: templates/repo/diff/compare.tmpl [hunk 13]
```diff
@@ -70,7 +70,7 @@
 				</div>
 				<div class="scrolling menu reference-list-menu base-branch-list">
 					{{range .Branches}}
-						<div class="item{{if eq $.BaseBranch .}}selected{{end}}" data-url="{{$.RepoLink}}/compare/{{PathEscapeSegments .}}{{$.CompareSeparator}}{{if not $.PullRequestCtx.SameRepo}}{{PathEscape $.HeadUser.Name}}/{{PathEscape $.HeadRepo.Name}} :{{end}}{{PathEscapeSegments $.HeadBranch}}">{{$BaseCompareName}}:{{.}}</div>
+						<div class="item{{if eq $.BaseBranch .}}selected{{end}}" data-url="{{$.RepoLink}}/compare/{{PathEscapeSegments .}}{{$.CompareSeparator}}{{if not $.PullRequestCtx.SameRepo}}{{PathEscape $.HeadUser.Name}}/{{PathEscape $.HeadRepo.Name}}:{{end}}{{PathEscapeSegments $.HeadBranch}}">{{$BaseCompareName}}:{{.}}</div>
```
Comment (zeripath) [inline templates/repo/diff/compare.tmpl]: You've missed the space

```suggestion
						<div class="item{{if eq $.BaseBranch .}} selected{{end}}" data-url="{{$.RepoLink}}/compare/{{PathEscapeSegments .}}{{$.CompareSeparator}}{{if not $.PullRequestCtx.SameRepo}}{{PathEscape $.HeadUser.Name}}/{{PathEscape $.HeadRepo.Name}}:{{end}}{{PathEscapeSegments $.HeadBranch}}">{{$BaseCompareName}}:{{.}}</div>
```

File: templates/repo/diff/compare.tmpl [hunk 14]
```diff
@@ -90,7 +90,7 @@
 				</div>
 				<div class="scrolling menu reference-list-menu base-tag-list" style="display: none">
 					{{range .Tags}}
-						<div class="item{{if eq $.BaseBranch .}}selected{{end}}" data-url="{{$.RepoLink}}/compare/{{PathEscapeSegments .}}{{$.CompareSeparator}}{{if not $.PullRequestCtx.SameRepo}}{{PathEscape $.HeadUser.Name}}/{{PathEscape $.HeadRepo.Name}} :{{end}}{{PathEscapeSegments $.HeadBranch}}">{{$BaseCompareName}}:{{.}}</div>
+						<div class="item{{if eq $.BaseBranch .}}selected{{end}}" data-url="{{$.RepoLink}}/compare/{{PathEscapeSegments .}}{{$.CompareSeparator}}{{if not $.PullRequestCtx.SameRepo}}{{PathEscape $.HeadUser.Name}}/{{PathEscape $.HeadRepo.Name}}:{{end}}{{PathEscapeSegments $.HeadBranch}}">{{$BaseCompareName}}:{{.}}</div>
```
Comment (zeripath) [inline templates/repo/diff/compare.tmpl]:
```suggestion
						<div class="item{{if eq $.BaseBranch .}} selected{{end}}" data-url="{{$.RepoLink}}/compare/{{PathEscapeSegments .}}{{$.CompareSeparator}}{{if not $.PullRequestCtx.SameRepo}}{{PathEscape $.HeadUser.Name}}/{{PathEscape $.HeadRepo.Name}}:{{end}}{{PathEscapeSegments $.HeadBranch}}">{{$BaseCompareName}}:{{.}}</div>
```

File: templates/repo/diff/compare.tmpl (hunk 13 above)
Comment (kecrily) [reply templates/repo/diff/compare.tmpl]: It seems that you are not viewing the latest version.
image

**Comment by silverwind** [comment]
Can you get the tests to pass please? Then I'm happy to review.

File: templates/repo/issue/list.tmpl [hunk 15]
```diff
@@ -14,13 +14,13 @@
 					{{if .PageIsIssueList}}
 						<a class="ui green button" href="{{.RepoLink}}/issues/new{{if .NewIssueChooseTemplate}}/choose{{end}}">{{.locale.Tr "repo.issues.new"}}</a>
 					{{else}}
-						<a class="ui green button {{if not .PullRequestCtx.Allowed}}disabled{{end}}" href="{{if .PullRequestCtx.Allowed}}{{.Repository.Link}}/compare/{{.Repository.DefaultBranch | PathEscapeSegments}}...{{if ne .Repository.Owner.Name .PullRequestCtx.BaseRepo.Owner.Name}}{{PathEscape .Repository.Owner.Name}}:{{end}}{{.Repository.DefaultBranch | PathEscapeSegments}}{{end}}">{{.locale.Tr "repo.pulls.new"}}</a>
+						<a class="ui green button{{if not .PullRequestCtx.Allowed}} disabled{{end}}" href="{{if .PullRequestCtx.Allowed}}{{.Repository.Link}}/compare/{{.Repository.DefaultBranch | PathEscapeSegments}}...{{if ne .Repository.Owner.Name .PullRequestCtx.BaseRepo.Owner.Name}}{{PathEscape .Repository.Owner.Name}}:{{end}}{{.Repository.DefaultBranch | PathEscapeSegments}} {{end}}">{{.locale.Tr "repo.pulls.new"}}</a>
```
Comment (zeripath) [inline templates/repo/issue/list.tmpl]:
```suggestion
						<a class="ui green button{{if not .PullRequestCtx.Allowed}} disabled{{end}}" href="{{if .PullRequestCtx.Allowed}}{{.Repository.Link}}/compare/{{.Repository.DefaultBranch | PathEscapeSegments}}...{{if ne .Repository.Owner.Name .PullRequestCtx.BaseRepo.Owner.Name}}{{PathEscape .Repository.Owner.Name}}:{{end}}{{.Repository.DefaultBranch | PathEscapeSegments}}{{end}}">{{.locale.Tr "repo.pulls.new"}}</a>
```

File: templates/repo/issue/list.tmpl [hunk 16]
```diff
@@ -14,13 +14,13 @@
...
 					{{end}}
 				</div>
 			{{else}}
 				{{if not .PageIsIssueList}}
 					<div class="column right aligned">
-						<a class="ui green button {{if not .PullRequestCtx.Allowed}}disabled{{end}}" href="{{if .PullRequestCtx.Allowed}}{{.PullRequestCtx.BaseRepo.Link}}/compare/{{.PullRequestCtx.BaseRepo.DefaultBranch | PathEscapeSegments}}...{{if ne .Repository.Owner.Name .PullRequestCtx.BaseRepo.Owner.Name}}{{PathEscape .Repository.Owner.Name}}:{{end}}{{.Repository.DefaultBranch | PathEscapeSegments}}{{end}}">{{$.locale.Tr "action.compare_commits_general"}}</a>
+						<a class="ui green button{{if not .PullRequestCtx.Allowed}} disabled{{end}}" href="{{if .PullRequestCtx.Allowed}}{{.PullRequestCtx.BaseRepo.Link}}/compare/{{.PullRequestCtx.BaseRepo.DefaultBranch | PathEscapeSegments}}...{{if ne .Repository.Owner.Name .PullRequestCtx.BaseRepo.Owner.Name}}{{PathEscape .Repository.Owner.Name}}:{{end}}{{.Repository.DefaultBranch | PathEscapeSegments}} {{end}}">{{$.locale.Tr "action.compare_commits_general"}}</a>
```
Comment (zeripath) [inline templates/repo/issue/list.tmpl]:
```suggestion
						<a class="ui green button{{if not .PullRequestCtx.Allowed}} disabled{{end}}" href="{{if .PullRequestCtx.Allowed}}{{.PullRequestCtx.BaseRepo.Link}}/compare/{{.PullRequestCtx.BaseRepo.DefaultBranch | PathEscapeSegments}}...{{if ne .Repository.Owner.Name .PullRequestCtx.BaseRepo.Owner.Name}}{{PathEscape .Repository.Owner.Name}}:{{end}}{{.Repository.DefaultBranch | PathEscapeSegments}}{{end}}">{{$.locale.Tr "action.compare_commits_general"}}</a>
```

**Comment by zeripath** [comment]
There were two more mistakenly added spaces.

**Comment by wxiaoguang** [comment]
I have objection to this change, and suggest to stick to `class="c1 {{if $var}}c2{{end}}"`

The readability and maintainability  should be applied to the code which is read by developers, but not for the generated outputs.

The template code is the code for developers, while the generated HTML are only for browsers.

The `class="c1 {{if $var}}c2{{end}}"` style is clearer for developers and more intuitive, and the generated HTML also makes browsers happy.

Think about a more complex case:

* `class="ui menu item {{if $show}}show{{end}} {{if $warn}}warn{{end}} {{if $error}}error{{end}}"`
* --vs--
* `class="ui menu item{{if $show}} show{{end}}{{if $warn}} warn{{end}}{{if $error}} error{{end}}"`

The first style make it clearer to see each CSS class name with its `{{if}}` block.

**Comment by zeripath** [comment]
I do agree with @wxiaoguang here - I know we've done a lot of work elsewhere getting rid of the extra spaces but I'm just not sure it's worth it given the confusion and potential for bugs it adds. 

Who is looking at the HTML anyway? We'd potentially be better off passing the output of our HTML renderer through a lightweight html parser that would minimize and remove the extra spaces instead.

**Comment by silverwind** [comment]
Not sure if postprocessing the HTML is worth it, it'll impact performance for neglible gain.

BTW, In JS land, there is the [classnames](https://www.npmjs.com/package/classnames) module that's used to join together class names, maybe something similar could be done for go templates as well, thought it'll never be as elegant as the JS solution.

**Comment by wolfogre** [comment]
Now that #23298 has been merged, I suggest closing this PR. @kecrily, thank you for your effort; it prompted us to think more deeply..
